internal/cmd/generate-txt-testdata/main.go

```diff
@@ -85,35 +83,5 @@ func tryParse(filePath string) (string, error) {
 	if err != nil {
 		return "", err
 	}
-
-	validator, err := protovalidate.New()
-	if err != nil {
-		return "", err
-	}
-
-	options := protoyaml.UnmarshalOptions{
-		Validator: validator,
-		Path:      filePath,
-	}
-
-	switch {
-	case strings.HasSuffix(filePath, ".proto2test.yaml"):
-		testCase := &testv1.Proto2Test{}
-		err = options.Unmarshal(data, testCase)
-	case strings.HasSuffix(filePath, ".proto3test.yaml"):
-		testCase := &testv1.Proto3Test{}
-		err = options.Unmarshal(data, testCase)
-	case strings.HasSuffix(filePath, ".const.yaml"):
-		testCase := &testv1.ConstValues{}
-		err = options.Unmarshal(data, testCase)
-	case strings.HasSuffix(filePath, ".validate.yaml"):
-		testCase := &testv1.ValidateTest{}
-		err = options.Unmarshal(data, testCase)
-	default:
-		return "", fmt.Errorf("unknown file type: %s", filePath)
-	}
-	if err != nil {
-		return err.Error(), nil
-	}
-	return "", nil
+	return golden.GenGoldenContent(filePath, data)
```

Comment:
I am super-confused by the previous formulation that you've replaced. It looks like it basically always returned the empty string on success, instead of `string(data)`. But the new formulation "round-trips" the file and returns the contents after re-marshaling.

All that intended? Mind adding some remarks in a comment or in the PR description to provide some guidance on some of this diff?

Reply:
Yes, I added a new 'class' of tests for valid .yaml files, that verifies the marshalled output is as expected.

Reply:
(basically, it validates that everything is normalized and in the right order for valid .yamls)